File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function(matrix, target) {
    let n = matrix.length-1;
    let m = matrix[0].length-1;
    let i = 0; // i indicates the current row
    let j = m; // j indicates the current column
    while(i <= n && j >= 0){
        if( matrix[i][j] == target){
            return true;
        }else{
            if( matrix[i][j] < target){
                i++;
            }else{
                j--;
            }
        }
    }
    return false;
};

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] nums, int tar) {
        int m = nums.length;
        int n = nums[0].length;
        int row = 0;
        int col = n-1;
            
            while(row < m && col >= 0){
               if(nums[row][col] == tar){
                return true;
               }else if(nums[row][col] > tar){
                col--;
               }else{
                row++;
               }
            }
        return false;
    }
}